walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -91,28 +137,38 @@
     }
 
     // time range objects are ordered, so you can compare them by order
-    for (int i=0; i<=eventsAsTimeRanges.size()-2; i++) {
+    for (int i = 0; i <= eventsAsTimeRanges.size()-2; i++) {
       TimeRange earlierTimeRange = eventsAsTimeRanges.get(i);
-      TimeRange laterTimeRange = eventsAsTimeRanges.get(i+1);
+      TimeRange laterTimeRange = eventsAsTimeRanges.get(i + 1);
 
       if (earlierTimeRange.overlaps(laterTimeRange)) { 
         int laterTRStart = laterTimeRange.start();
         int laterTREnd = laterTimeRange.end();
 
-        // case 1: "regular" overlapping events 
+        // CASE 1: "regular" overlapping events 
         if (earlierTimeRange.contains(laterTRStart) && (! earlierTimeRange.contains(laterTREnd))) {
           TimeRange newTimeRange = TimeRange.fromStartEnd(earlierTimeRange.start(), laterTREnd, false);
           unavailableTimeRanges.add(newTimeRange);
         } else if (earlierTimeRange.contains(laterTimeRange)) {
-          // case 2: nested events
+          // CASE 2: nested events
           unavailableTimeRanges.add(earlierTimeRange);
         }
       } else {
-        // case 3: non-overlapping events
-        unavailableTimeRanges.add(earlierTimeRange);
+        // CASE 3: non-overlapping events
+        if (unavailableTimeRanges.size() >= 1) {
+          // make sure that last unavailableTimeRanges object does not overlap
+          if (!(unavailableTimeRanges.get(unavailableTimeRanges.size()-1).contains(laterTimeRange.end()))) {
+            unavailableTimeRanges.add(earlierTimeRange);
+          }
```

Comment:
simplify if condition and add spaces

`if (!unavailableTimeRanges.get(unavailableTimeRanges.size() - 1).contains(laterTimeRange.end())) {`